frontend/src/tests/user-permission-tests.js

```diff
@@ -14,42 +14,127 @@ import { databaseSeeder } from "./setup";
 export function userPermissionsTests(api) {
     const { apiGET, apiPOST } = api;
 
-    const newUserData = {
+    let defaultUser;
+
+    const newInstructorOnlyUserData = {
         utorid: "userxx",
         roles: ["instructor"],
     };
 
+    const newTAOnlyUserData = {
+        utorid: "tauser",
+        roles: ["ta"],
+    };
+
+    const newTAInstructorUserData = {
+        utorid: "tainstuser",
+        roles: ["ta", "instructor"],
+    };
+
+    /**
+     * Switch current active user to the user represented by the given user data. If given user does not exist, one will
+     * be created.
+     * @param userData the user to be set to the active user
+     * @returns {Promise<void>}
+     */
+    async function switchToTargetUser(userData) {
+        let resp = await apiPOST("/debug/users", userData);
+        expect(resp).toHaveStatus("success");
+
+        // Set the active user to the newly-created user
+        resp = await apiPOST("/debug/active_user", { id: resp.payload.id });
+        expect(resp).toHaveStatus("success");
+    }
+
+    /**
+     * Restores the active user to the default user (the user logged during test setup in beforeAll).
+     *
+     * @returns {Promise<void>}
+     */
+    async function restoreDefaultUser() {
+        let respSwitchBackUser = await apiPOST(
+            `/debug/active_user`,
+            defaultUser
+        );
+        expect(respSwitchBackUser).toHaveStatus("success");
+    }
+
     beforeAll(async () => {
         await databaseSeeder.seed(api);
+        const respOriginalUser = await apiGET(`/debug/active_user`);
+        expect(respOriginalUser).toHaveStatus("success");
+        defaultUser = {
+            utorid: respOriginalUser.payload.utorid,
+            roles: respOriginalUser.payload.roles,
+        };
     }, 30000);
 
+    afterEach(async () => {
+        await restoreDefaultUser();
```

Comment:
It seems weird to have an `afterEach` with no `beforeEach`. It makes it hard to follow exactly what's going on in each test. Just repeat the one line each time you need it.